src/main/scala/io/iohk/ethereum/network/p2p/MessageDecoders.scala

```diff
@@ -38,45 +36,72 @@ object NetworkMessageDecoder extends MessageDecoder {
 // scalastyle:off
 object EthereumMessageDecoder extends MessageDecoder {
 
-  override def fromBytes(msgCode: Int, payload: Array[Byte], protocolVersion: Version): Message =
-    (protocolVersion, msgCode) match {
-      //wire protocol
-      case (_, Hello.code) => payload.toHello
-
-      //FIXME: I still have an issue with protocolVersion, we are use PV62 and PV63 and code names for Status and NewBlock
-      // suggest that there is PV64, but there isn't
-      //common
-      case (_, Status.code63 | Status.code64) => payload.toStatus(msgCode)
-      case (_, SignedTransactions.code) => payload.toSignedTransactions
-      case (_, NewBlock.code63 | NewBlock.code64) => payload.toNewBlock(msgCode)
-
-      case (PV61, t) => handlePV61(t, payload)
+  override def fromBytes(msgCode: Int, payload: Array[Byte], protocolVersion: Version): Message = {
+    protocolVersion match {
+      case PV61 => handlePV61(msgCode, payload)
+      case PV62 => handlePV62(msgCode, payload)
+      case PV63 => handlePV63(msgCode, payload)
+      case PV64 => handlePV64(msgCode, payload)
```

Comment:
Very minor optimization: higher PVs will be matched more often, so they could matched first